log/root.go

```diff
@@ -2,22 +2,26 @@ package log
 
 import (
 	"os"
-	"sync/atomic"
+	"sync"
 
 	"golang.org/x/exp/slog"
 )
 
 var (
-	root = new(atomic.Value)
+	rootMu sync.Mutex
+	root logger
 )
 
 func init() {
 	defaultLogger := &logger{slog.New(DiscardHandler())}
-	root.Store(defaultLogger)
+	SetDefault(defaultLogger)
 }
 
 func SetDefault(l Logger) {
-	root.Store(l)
+	rootMu.Lock()
+	defer rootMu.Unlock()
+	root := l
```

Comment:
This redefines root. Did you mean '=' not ':=' ?

Reply:
Yeah good catch.